subgraphs/aave-forks/protocols/aave-v3/src/mapping.ts

```diff
@@ -791,18 +794,22 @@ function getBalanceTransferAmount(event: ethereum.Event): BigInt {
   );
   for (let i = 0; i < logs.length; i++) {
     const thisLog = logs[i];
-    if (thisLog.logIndex.le(eventLogIndex)) {
-      // skip event with logIndex < event.logIndex
-      continue;
-    }
-    // topics[0] - signature
-    const logSignature = thisLog.topics[0];
-    if (thisLog.address == event.address && logSignature == eventSignature) {
-      const UINT256_UINT256 = "(uint256,uint256)";
-      const decoded = ethereum.decode(UINT256_UINT256, thisLog.data)!.toTuple();
-      btAmount = decoded[0].toBigInt();
-
-      break;
+    if (thisLog.topics.length > INT_ZERO) {
+      if (thisLog.logIndex.le(eventLogIndex)) {
+        // skip event with logIndex < event.logIndex
+        continue;
+      }
+      // topics[0] - signature
+      const logSignature = thisLog.topics[0];
+      if (thisLog.address == event.address && logSignature == eventSignature) {
+        const UINT256_UINT256 = "(uint256,uint256)";
+        const decoded = ethereum
+          .decode(UINT256_UINT256, thisLog.data)!
+          .toTuple();
+        btAmount = decoded[0].toBigInt();
```

Comment:
should we check if the index exists before calling it? Great bug fix here, this was not easy to find

Reply:
checking if `log.data` is decoded into the given signature, if yes, it will have this index